### PR TITLE
YLP-695 Fix legend overlap in daily bolus graph

### PR DESCRIPTION
**Pull request by yvon-dblg**
Question: Do I need to update the changelog ?

Please, squash me !

image

### CODE REVIEW

File: packages/tideline/plugins/blip/chartdailyfactory.js [hunk 1]
```diff
@@ -89,70 +90,91 @@ function chartDailyFactory(parentElement, tidelineData, options = {}) {
 
   // top x-axis pool
   /** @type {Pool} */
-  const poolXAxis = chart.newPool()
-    .id('poolXAxis', chart.poolGroup)
-    .label('')
-    .labelBaseline(options.labelBaseline);
-  poolXAxis.index(chart.pools.indexOf(poolXAxis))
+  const poolXAxis = new Pool(chart);
+  chart.addPool(poolXAxis);
```
Comment (franck-fourel) [inline packages/tideline/plugins/blip/chartdailyfactory.js:94]: the way we create those objects are really strange.
I create a pool using the chart container
```
  const poolXAxis = new Pool(chart);
```
and then I add this new created pool to the list of pools of the original object
```
  chart.addPool(poolXAxis);
```

**Review by franck-fourel — APPROVED** [review]
Work as expected
couple of questions

File: packages/tideline/js/pool.js [hunk 2]
```diff
@@ -17,7 +17,7 @@
 
 import _ from 'lodash';
 
-import legend from './plot/util/legend';
+import legendDefs from './plot/util/legend';
```
[the remark this answers is not in the document]
Comment (yvon-dblg) [reply packages/tideline/js/pool.js:20]: Not sure to remember why, it's been a long time.
I think it's to know theses are definitions of legends.

File: packages/tideline/plugins/blip/chartdailyfactory.js (hunk 1 above)
Comment (yvon-dblg) [reply packages/tideline/plugins/blip/chartdailyfactory.js:94]: Yes I know.
I change all theses little things steps by steps.
Someday I hope ....

**Pull request merged**